file-system.ts

```diff
@@ -132,7 +133,8 @@ export class FileSystem implements IFileSystem {
 
 		return future;
 	}
-
+	
+	@decorators.promisify("fs")
```

Comment:
I couldn't find a way to skip the "fs" parameter. The problem is that method decorators are executed before class decorators, so at the moment when the method is promisified, our injector still does not know anything about "fs"